### PR TITLE
CCIPReader pricing tests

### PR DESCRIPTION
**Pull request by asoliman92**
Add functionality to bind more contracts to different chains (using same simulated backend for simplicity) in ccipreader tester.

* LinkPriceUSD test
* GetMedianDataAvailabilityGasConfig
* GetChainFeePriceUpdates
* GetWrappedNativeTokenPriceUSD

### CODE REVIEW

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 1]
```diff
@@ -605,4 +913,13 @@ type testSetupData struct {
 	lp           logpoller.LogPoller
 	cl           client.Client
 	reader       ccipreaderpkg.CCIPReader
+	extendedCR   contractreader.Extended
+}
+
+func UBigInt(i uint64) *big.Int {
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: private

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 2]
```diff
@@ -605,4 +913,13 @@ type testSetupData struct {
...
+	return new(big.Int).SetUint64(i)
+}
+
+func E18Mult(amount uint64) *big.Int {
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: private

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 3]
```diff
@@ -547,26 +828,52 @@ func testSetup(
...
+
+	for chain, bindings := range toMockBindings {
 		m := readermocks.NewMockContractReaderFacade(t)
 		m.EXPECT().Bind(ctx, bindings).Return(nil)
 		ecr := contractreader.NewExtendedContractReader(m)
 		err = ecr.Bind(ctx, bindings)
 		require.NoError(t, err)
+		if _, ok := otherCrs[chain]; ok {
+			t.Fatalf("chain %d already exists", chain)
+		}
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: require.False(t, otherCrs[chain])

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 4]
```diff
@@ -479,32 +503,289 @@ func TestCCIPReader_Nonces(t *testing.T) {
...
+	sb.Commit()
+
+	gas, err := feeQuoter.GetDestinationChainGasPrice(&bind.CallOpts{}, uint64(chainS1))
+	require.NoError(t, err)
+	require.Equal(t, defaultGasPrice.ToInt(), gas.Value)
+
+	return feeQuoter
+}
+
+func DefaultFeeQuoterDestChainConfig() fee_quoter.FeeQuoterDestChainConfig {
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: private

**Review by makramkd — APPROVED** [review]

**Review by 0xAustinWang — APPROVED** [review]

**Review by dimkouv — APPROVED** [review]

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 5]
```diff
@@ -479,32 +503,289 @@ func TestCCIPReader_Nonces(t *testing.T) {
...
 	readerChain,
 	destChain cciptypes.ChainSelector,
 	onChainSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum,
 	cfg evmtypes.ChainReaderConfig,
-	otherBindings map[cciptypes.ChainSelector][]types.BoundContract,
+	toBindContracts map[cciptypes.ChainSelector][]types.BoundContract,
+	toMockBindings map[cciptypes.ChainSelector][]types.BoundContract,
+	bindTester bool,
+	simulatedBackend *simulated.Backend,
+	auth *bind.TransactOpts,
```
Comment (winder) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go:787]: might be worth having a parameter object next time this changes so you don't have to go and update old tests

**Review by winder — APPROVED** [review]

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go (hunk 5 above)
Comment (asoliman92) [reply core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go:787]: Yeah. I'll try to refactor this in a subsequent PR to also change older tests to use real contracts.

**Review by 0xnogo — APPROVED** [review]

**Pull request merged**